### PR TITLE
[REG-2004] Update async load pattern for sequences AND fix prod compile

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Updates the async pattern for loading sequences to use Coroutines due to issues with API calls on non-main thread
- Fixes compilation/functionality issues in the production build #if branches of the code

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

**Review by batu — APPROVED** [review]
Looks good!

File: src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceManager.cs [hunk 1]
```diff
@@ -42,15 +50,7 @@ public static RGSequenceManager GetInstance()
...
-        if (_loadingSequences == false)
-        {
-            // not perfectly thread safe.. but close enough for this case
-            _loadingSequences = true;
-            new Thread(() =>
-            {
-                _loadedSequences = ResolveSequenceFiles();
-            }).Start();
-        }
+        StartCoroutine(ResolveSequenceFiles());
```
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceManager.cs:53]: nit: Is this LoadSequences called anywhere else (I can't tell from the GitHub PR review). If not, we should just call this code directly in start.

**Review by vontell — APPROVED** [review]

File: src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceManager.cs [hunk 2]
```diff
@@ -42,15 +50,7 @@ public static RGSequenceManager GetInstance()
      */
     public void LoadSequences()
     {
-        if (_loadingSequences == false)
```
[the remark this answers is not in the document]
Comment (RG-nAmKcAz) [reply src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceManager.cs:45]: eesh, i didn't even see this comment the way the page updated for me when i saw aaron's update.. i'll be sure to take these out in a future PR

File: src/gg.regression.unity.bots/Plugins/SimpleFileBrowser/Sprites/SimpleFileBrowserSpriteAtlas.spriteatlas [hunk 3]
```diff
@@ -21,7 +21,7 @@ SpriteAtlas:
       crunchedCompression: 0
       sRGB: 1
     platformSettings:
-    - serializedVersion: 2
```
[the remark this answers is not in the document]
Comment (RG-nAmKcAz) [reply src/gg.regression.unity.bots/Plugins/SimpleFileBrowser/Sprites/SimpleFileBrowserSpriteAtlas.spriteatlas:24]: yes/no.. i didn't intend to change it, but this keeps popping up consistently for every pr so i went ahead and committed it to hopefully not have to see it anymore

File: src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceManager.cs (hunk 1 above)
Comment (RG-nAmKcAz) [reply src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceManager.cs:53]: it is called in multiple locations yes

**Pull request merged**